File: secondlargest.py

first_largest = 0
second_largest = 0

listmain = [12,1,34,78,89,100,2,4,100]

for item in listmain:
    if item > first_largest:
        second_largest = first_largest
        first_largest = item
    
    elif item > second_largest and item != first_largest:
        second_largest = item


print(second_largest)

File: secondlargest.java
public class secondlargest {
    public static void main(String[] args) {
        int first_largest = 0;
        int second_largest = 0;

        int [] arr = {2,45,65,7,21,90,90};

        for (int item : arr) {
            if (item > first_largest){
                second_largest = first_largest;
                first_largest = item;
            }
            
            else if(item > second_largest && item != first_largest){
                second_largest = item;
            }
        }
    
        System.out.println(second_largest);
    }
    
}
